vllm/v1/engine/core.py

```diff
@@ -302,13 +302,14 @@ def step_with_batch_queue(
         # Note that this is not blocking.
         if not self.batch_queue.full():
             scheduler_output = self.scheduler.schedule()
-            if scheduler_output.total_num_scheduled_tokens > 0:
+            if scheduler_output.total_num_scheduled_tokens > 0 or scheduler_output.finished_req_ids:
                 future = self.model_executor.execute_model(scheduler_output)
                 self.batch_queue.put_nowait(
                     (future, scheduler_output))  # type: ignore
 
         scheduled_batch = (scheduler_output is not None
-                           and scheduler_output.total_num_scheduled_tokens > 0)
+                           and (scheduler_output.total_num_scheduled_tokens > 0
+                                or scheduler_output.finished_req_ids))
```

Comment:
Instead of these lines, just add `scheduled_batch = True` right after `self.batch_queue.put_nowait(...)`?